signer/src/transaction_signer.rs

```diff
@@ -796,13 +1110,24 @@ where
             return Err(Error::MissingStateMachine);
         };
 
+        let mut frost_coordinator = if let StateMachineId::RotateKey(_, _) = state_machine_id {
+            self.wsts_frost_state_machines.get_mut(&state_machine_id)
+        } else {
+            None
+        };
+
         let outbound_messages = state_machine.process(msg).map_err(Error::Wsts)?;
 
         for outbound_message in outbound_messages.iter() {
             // The WSTS state machine assume we read our own messages
             state_machine
                 .process(outbound_message)
                 .map_err(Error::Wsts)?;
+
+            if let Some(frost_coordinator) = &mut frost_coordinator {
```

Comment:
nit: redundant &mut
```suggestion
            if let Some(frost_coordinator) = frost_coordinator {
```

Reply:
nah it was needed due to move problems otherwise since it's in a loop -- but maybe the syntax in [1e74dbf](https://github.com/stacks-network/sbtc/pull/1285/commits/1e74dbf9e036d33d3a58a3da4b7f17c502e31a6f) is nicer